lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -677,4 +677,168 @@ protected static function do_opt_in_into_settings( &$context ) {
 
 		unset( $context['appearanceTools'] );
 	}
+
+	/**
+	 * Given a settings array, it returns the generated rulesets
+	 * for the preset classes.
+	 *
+	 * @param array  $settings Settings to process.
+	 * @param string $selector Selector wrapping the classes.
+	 * @param array  $origins  List of origins to process.
+	 * @return string The result of processing the presets.
+	 */
+	protected static function compute_preset_classes( $settings, $selector, $origins ) {
```

Comment:
This code is producing:

```css
.has-small-font-size {
  --wp--user--preset--font-size: var(--wp--preset--font-size--small);
  font-size: var(--wp--preset--font-size--small);
}
```

If I understood the proposal, shouldn't it be?

```css
.has-small-font-size {
  --wp--user--preset--font-size: var(--wp--preset--font-size--small);
  font-size: var(--WP--USER--PRESET--FONT-SIZE);
}
```

Reply:
I don't actually know if these are super useful for font sizes compared to colors, as font sizes are more straightforward and there is no `.has-font-size` type class to latch on to and set these like there is for `.has-background`.

Reply:
given that at this point `--wp--preset--font-size--small` are both the same at this point it really makes no difference, the use of the `--wp--user--preset--font-size` is only critical at the points where we need to make sure it takes precedence, eg. with a table font declaration:
```css
.wp-block-table > table {
    font-size: var(--wp--user--preset--font-size,var(--wp--preset--font-size--large));
}
```
However, to avoid confusion I have updated it to use `--wp--user--preset--font-size` in both places.